packages/core/src/lib/relay/index.ts

```diff
@@ -73,12 +59,13 @@ class Relay extends GossipSub implements IRelay {
       globalSignaturePolicy: SignaturePolicy.StrictNoSign,
       fallbackToFloodsub: false,
     });
+
     super(components, options);
     this.multicodecs = constants.RelayCodecs;
 
     this.observers = new Map();
 
-    this.pubSubTopic = options?.pubSubTopic ?? DefaultPubSubTopic;
+    this.options = options ?? {};
```

Comment:
you end up applyuing the default pubsub topic and every usage of `this.option`. This is cumbersome and prone to developer error.

Apply the default values in the constructor:
```
this.options = Object.assign({pubSubTopic: DefaultPubSubTopic}, options, {})
```

Or same logic in a getter `get options(): ProtocolOptions`

Or create a class that handles the `ProtocolOptions` and automatically apply default values